### PR TITLE
Instance relators

### PR DESCRIPTION
**Pull request by davidgrove73**
this time with the right branch

copyright_date is added as a property in instances
copyright_agent is a relator (role: copyright_holder) between instance and agent object

publishing info (including publisher) is now handled exclusively by the provider class

@xenma @naderak @jolf

### CODE REVIEW

File: app/models/instance.rb [hunk 1]
```diff
@@ -54,6 +55,22 @@ def set_rights_metadata
...
+  end
+
+  def delete_copyright_relations
+    if self.copyright_status != 'agent'
+      select_relators('cph').each do |rel|
+        rel.delete
+      end
+    end
+  end
+
```
Comment (davidgrove73) [inline app/models/instance.rb]: i should delete this function again. It is not used

File: app/models/relator.rb [hunk 2]
```diff
@@ -31,7 +31,7 @@ def agent_id=(id)
   end
 
   def agent_id
-    self.agent.try(:id)
+    self.agent.try(:id) if agent.present?
```
Comment (xenma) [inline app/models/relator.rb]: Since we use the try function, isn't the if statement redundant? (mostly a question)

**Comment by xenma** [comment]
Cannot delete an organization: When I edit an instance with an organization, and delete the organization, it actually keeps the old one.

File: app/views/authority/organizations/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<h1><%= t('.title') %></h1>
+
+<%= render 'form' %>
+
+<%= link_to t(:show), @authority_object %> |
+<%= link_to t(:back), authority_organization_path%>
```
Comment (xenma) [inline app/views/authority/organizations/new.html.erb]: authority_organizations_path

File: app/models/relator.rb (hunk 2 above)
Comment (davidgrove73) [inline app/models/relator.rb]: it propably is

**Comment by davidgrove73** [comment]
I fixed the problem with deleting publisher

**Pull request merged**